### PR TITLE
[Bug] Fix weird animation of toolbar 

### LINKED ISSUE #12659: Weird animation of toolbar when switching from Payments to Manage card reader screen
**Describe the bug**
A clear and concise description of what the bug is. Please be as descriptive as possible; issues lacking detail, or for any other reason than to report a bug, may be closed without action.
The same issue happens to other screens when we assumebly going from activity toolbar to the local toolbar

Please check the video for better understanding

**To Reproduce**
1. Go to More -> Payments
2. Open Manager card reader -> Notice that the whole list goes down due to double toolbar
3. Go back -> notice Toolbar jumping

**Screenshots**

image

https://github.com/user-attachments/assets/e30ca54e-28e1-4ef8-a5e6-a32bc6ce3eab

### PR DESCRIPTION
**Pull request by backwardstruck**
Closes: #12659  

### Description  

This PR addresses the issue of inconsistent toolbar behavior and weird animation glitches when transitioning between the Payments Hub screen and `CardReaderDetailFragment`, `CardReaderManualsFragment`. The solution involves making these fragments use a local toolbar while hiding the activity toolbar, thereby ensuring a consistent and smooth navigation experience.

### Steps to Reproduce  

1. Navigate to the "Payments" screen in More tab.
2. Click on "Manage Card Reader".
3. Observe that the toolbar now belongs to the fragment itself and functions correctly without any animation glitches.
4. Navigate to the "Card Reader Manuals" section.
5. Observe the same behavior with a correctly functioning local toolbar.
6. [On a device with NFC] Tap on Share Feedback in TAP TO PAY
7. Submit test feedback
8. Observe the same behavior with a correctly functioning local toolbar.

### Testing Information  

- Devices used: tablet emulator
- Edge cases: Testing on both phone and tablet devices

### The Tests That Have Been Performed  

- Manual navigation testing between `CardReaderDetailFragment`, `CardReaderManualsFragment`, and other related fragments.
- Regression testing to verify that no new issues have been introduced.
- Light and dark mode

- [X] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.  
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.  
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsFragment.kt [hunk 1]
```diff
@@ -27,7 +39,23 @@
...
+                                            painter = painterResource(R.drawable.ic_back_24dp),
+                                            contentDescription = stringResource(R.string.back)
+                                        )
+                                    }
+                                }
+                            )
+                        }
+                    ) {
+                        ManualsScreen(cardReaderManualsViewModel = viewModel)
+                    }
```
[the remark this answers is not in the document]
Comment (backwardstruck) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsFragment.kt]: Fixed

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsFragment.kt [hunk 2]
```diff
@@ -27,7 +40,24 @@ class CardReaderManualsFragment : BaseFragment() {
             setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
             setContent {
                 WooThemeWithBackground {
-                    ManualsScreen()
+                    Scaffold(
+                        topBar = {
+                            TopAppBar(
```
Comment (backwardstruck) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsFragment.kt]: Should I add a Compose UI test for this code? I would probably need to move it into the `ManualsScreen` class in that case.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/238">20.6</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsFragment.kt (hunk 2 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsFragment.kt]: I actually never did that. If they run on JVM and fast and easy to write then why not

Also I would suggest:
* To move all the compose code to the separate `ManualsScreen`
* Rename `ManualsScreen` to `CardReaderManualsScreen` for consistency

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsFragment.kt [hunk 3]
```diff
@@ -18,6 +28,9 @@ class CardReaderManualsFragment : BaseFragment() {
     override fun getFragmentTitle() = resources.getString(R.string.settings_card_reader_manuals)
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsFragment.kt]: Do we still need that?

**Review by kidinov — COMMENTED** [review]
LGTM!

I left a couple of comments please take a look.

Also, I would consider doing the same for "TAP TO PAY" -> Share Feedback, as we have the same bug there. We will fix this for the payments section completely then. wdyt?

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsFragment.kt (hunk 3 above)
Comment (backwardstruck) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsFragment.kt]: Nope.

**Comment by backwardstruck** [comment]
> LGTM!
> 
> I left a couple of comments please take a look.
> 
> Also, I would consider doing the same for "TAP TO PAY" -> Share Feedback, as we have the same bug there. We will fix this for the payments section completely then. wdyt?

I updated the `FeedbackSurveyFragment` to use its own toolbar and ensure the activity's toolbar is hidden when the fragment is displayed.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/PaymentsHubViewModel.kt [hunk 4]
```diff
@@ -567,8 +567,8 @@ class PaymentsHubViewModel @Inject constructor(
         ) : PaymentsHubEvents()
 
         data object NavigateToOrderCreationScreen : PaymentsHubEvents()
-        data object NavigateToTapTooPaySummaryScreen : PaymentsHubEvents()
-        data object NavigateToTapTooPaySurveyScreen : PaymentsHubEvents()
+        data object NavigateToTapToPaySummaryScreen : PaymentsHubEvents()
+        data object NavigateToTapToPaySurveyScreen : PaymentsHubEvents()
```
Comment (backwardstruck) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/hub/PaymentsHubViewModel.kt:571]: Fixed these typos.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsScreen.kt [hunk 5]
```diff
@@ -23,14 +28,45 @@ import androidx.compose.ui.res.stringResource
...
 import androidx.compose.ui.tooling.preview.Preview
 import androidx.lifecycle.viewmodel.compose.viewModel
+import androidx.navigation.NavController
 import com.woocommerce.android.R
+import com.woocommerce.android.ui.compose.theme.WooThemeWithBackground
 
 @Composable
-fun ManualsScreen(
-    cardReaderManualsViewModel: CardReaderManualsViewModel = viewModel()
+fun CardReaderManualsScaffold(
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsScreen.kt]: np: I think this is more of a "Screen" than scaffold in terms of how we call it. You can call them both `CardReaderManualsScreen`, but have the bottom one as private

**Review by kidinov — APPROVED** [review]
LGTM!

There are few lint warnings; some seem to make sense. Just pay attention to the background one - make sure that the actually background color won't be changed in both dark and light mode if you remove it

Also, I just wanted to validate with you that before, the completed screen was not fully white. I actually think that how you made it now is better, but I just to make sure that this is something you wanted to change

image

image

**Comment by backwardstruck** [comment]
>Also, I just wanted to validate with you that before, the completed screen was not fully white. I actually think that how you made it now is better, but I just to make sure that this is something you wanted to change

You're right @kidinov I didn't notice that in my testing. Because I change the top level view from `ScrollView` to `LinearLayout`. I agree it's OK since it still looks fine in light/dark mode and is still scrollable depending on the content.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsScreen.kt (hunk 5 above)
Comment (backwardstruck) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/manuals/CardReaderManualsScreen.kt]: Updated that.

**Pull request merged**